src/components/query-viewer/query-viewer.tsx

```diff
@@ -0,0 +1,369 @@
+   * If includeTrend, defines the period of the trend
+   */
+  @Prop() trendPeriod:
+    | "SinceTheBeginning"
+    | "LastYear"
+    | "LastSemester"
+    | "LastQuarter"
+    | "LastMonth"
+    | "LastWeek"
+    | "LastDay"
+    | "LastHour"
+    | "LastMinute"
+    | "LastSecond";
+  /**
+   * For timeline for remembering layout
+   */
+  @Prop() rememberLayout: boolean;
+  /**
+   * Orientation of the graph
+   */
+  @Prop() orientation: "Horizontal" | "Vertical";
+  /**
+   * Include spark line
+   */
+  @Prop() includeSparkline: boolean;
+  /**
+   * Include max and min
+   */
+  @Prop() includeMaxMin: boolean;
+  /**
+   * Theme for showing the graph
+   */
+  @Prop() theme: string;
+  /**
+   * Object type -> Query or DataProvider
+   */
+  @Prop() objectType: string;
+  /**
+   * True if it is external query
+   */
+  @Prop() isExternalQuery: boolean;
+  /**
+   * Allowing elements order to change
+   */
+  @Prop() allowElementsOrderChange: boolean;
+  /**
+   * If type== PivotTable or Table, if true will shrink the table
+   */
+  @Prop() autoResize: boolean;
+  /**
+   * If autoResize, in here select the type, Width, height, or both
+   */
+  @Prop() autoResizeType: "Both" | "Vertical" | "Horizontal";
+  /**
+   * Type of font
+   */
+  @Prop() fontFamily: string;
+  /**
+   * Font size
+   */
+  @Prop() fontSize: number;
+  /**
+   * Font Color
+   */
+  @Prop() fontColor: string;
+  /**
+   * Auto refresh group
+   */
+  @Prop() autoRefreshGroup: string;
+  /**
+   * Allowing or not Comlumn sort
+   */
+  @Prop() disableColumnSort: boolean;
+  /**
+   * Allow selection
+   */
+  @Prop() allowSelection: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XML
+   */
+  @Prop() exportToXML: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to HTML
+   */
+  @Prop() exportToHTML: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XLS
+   */
+  @Prop() exportToXLS: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XLSX
+   */
+  @Prop() exportToXLSX: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to PDF
+   */
+  @Prop() exportToPDF: boolean;
+  /**
+   * Title of the QueryViewer
+   */
+  @Prop() queryTitle: string;
+
+  private postData() {
+    const postBody = [];
+    for (const key in Object(this)) {
+      if (
+        Object(this)[key] != undefined &&
+        !this.propsNotToPost.includes(key)
+      ) {
+        postBody.push(
+          <input type="hidden" name={key} value={Object(this)[key]} />
+        );
+      }
+    }
+    postBody.push(<input type="hidden" name="Width" value={this.getWidth()} />);
+    postBody.push(
+      <input type="hidden" name="Height" value={this.getHeight()} />
+    );
+    postBody.push(
+      <input type="hidden" name="Elements" value={this.getElements()} />
+    );
+    postBody.push(
+      <input type="hidden" name="Parameters" value={this.getParameters()} />
+    );
+
+    return postBody;
```

Comment:
Nothing wrong with this approach, but wanted to share with you this more compact and declarative alternative:

```suggestion
    return [
      ...Object
        .keys(this)
        .filter(key => !this.propsNotToPost.includes(key))
        .map(key => <input type="hidden" name={key} value={this[key]} />),
      <input type="hidden" name="Width" value={this.getWidth()} />,
      <input type="hidden" name="Height" value={this.getHeight()} />,
      <input type="hidden" name="Elements" value={this.getElements()} />,
      <input type="hidden" name="Parameters" value={this.getParameters()} />
    ];
```